simple_salesforce/api.py

```diff
@@ -148,13 +152,17 @@ def __init__(
                     self.sf_instance += ':' + str(port)
             else:
                 self.sf_instance = instance
+            
+            # Only generate the headers wihtout logging in first
+            self._generate_headers()
```

Comment:
Since we don't actually perform a login here, we have to remember to generate the headers (usually done in the `_refresh_session` method).